pylint/reporters/reports_handler_mix_in.py

```diff
@@ -12,17 +13,19 @@ class ReportsHandlerMixIn:
     related methods for the main lint class
     """
 
-    def __init__(self):
-        self._reports = collections.defaultdict(list)
-        self._reports_state = {}
+    def __init__(self) -> None:
+        self._reports: DefaultDict = collections.defaultdict(list)
+        self._reports_state: Dict = {}
 
     def report_order(self):
         """Return a list of reports, sorted in the order
         in which they must be called.
         """
         return list(self._reports)
 
-    def register_report(self, reportid, r_title, r_cb, checker):
+    def register_report(
+        self, reportid: str, r_title: str, r_cb: Callable, checker: Any
```

Comment:
Based on the docstring of the only function that calls `register_report()` I think we can do `checker: IChecker`.
The `Callable` seems much harder to type and might be best handled when we get to typing `linter.reports`

Reply:
IChecker works :)

Reply:
Do you mean typing ``pylint/lint/report_functions.py`` ?

Reply:
Yes!